docs/source/developers/cpp/building.rst

```diff
@@ -41,6 +41,9 @@ Building requires:
   sufficient. For Windows, at least Visual Studio 2017 is required.
 * CMake 3.5 or higher
 * On Linux and macOS, either ``make`` or ``ninja`` build utilities
+* At least 1GB of RAM for a minimal build, 4GB for a minimal  
+  debug build with tests and 8GB for a full build using
+  :ref:`docker <docker>`.
```

Comment:
One thing that we've run into is that enabling the Unity build will frequently cause more memory to be required. https://issues.apache.org/jira/browse/ARROW-14555 is an example of that (it appears the VM had/was using 4GB when it was OOMkilled building unity). Maybe we should mention here that in ram-constrained environments that's one flag to turn off (ie. `CMAKE_UNITY_BUILD=OFF`)

Reply:
This is helpful. It seems that about 8GB is what is needed for most options, but that one can often use less. VM's with 1GB are significantly cheaper than VMs with 8GB